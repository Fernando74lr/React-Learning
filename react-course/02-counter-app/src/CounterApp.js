import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);
    console.log('~ counter:', counter);

    const handleAdd = () => {
        setCounter(counter + 1);
        // setCounter( (c) => c + 1); /* If for some reason we don't have acces to, in this case, 'counter' */
    };

    const handleSubstract = () => {
        setCounter(counter - 1);
    };

    const handleReset = () => {
        setCounter(value);
    };

    return (
        <div>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            <button onClick={ handleAdd } className={ 'green' }>+1</button>
            <button onClick={ handleReset } className={ 'red' }>Reset</button>
            <button onClick={ handleSubstract } className={ 'yellow' }>-1</button>
        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;
