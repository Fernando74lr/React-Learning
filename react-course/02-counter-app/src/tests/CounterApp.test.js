import CounterApp from "../CounterApp";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

describe('Tests in <CounterApp />', () => {

    const wrapper = shallow(<CounterApp />);

    test('should show <CounterApp /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should show subtitle sent by props', () => {
        const value = 100;

        const wrapper = shallow( <CounterApp value={value} />);

        const textPara = wrapper.find('h2').text().trim();

        expect(textPara).toBe(value.toString());
    });

    test('should increment counter in 1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('1');

    });

    test('should decrement counter in 1', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('0');

    });

    test('should put the default value with the btn reset', () => {
        const wrapper = shallow(<CounterApp value={ 50 } />);
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('50');

    })
    
    
});
