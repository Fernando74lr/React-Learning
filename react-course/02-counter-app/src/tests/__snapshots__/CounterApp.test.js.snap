// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CounterApp /> should show <CounterApp /> correctly 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    className="green"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="red"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="yellow"
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
